<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>zrmmethodeditor</class>
 <widget class="QWidget" name="zrmmethodeditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>432</width>
    <height>486</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <widget class="QFrame" name="btn_frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="leftMargin">
       <number>4</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item>
       <widget class="QToolButton" name="buttonLink">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonAllMethods">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="buttonApply">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonUndo">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="buttonDelete">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonMethodEdit">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="buttonNew">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonNewChild">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonCopyModel">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonCopyMethod">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_6">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="buttonLoad">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="buttonUnload">
        <property name="text">
         <string>...</string>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_7">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="param_widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="ZrmStagesEditor" name="stages_page">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
     <widget class="QWidget" name="link_page">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="ZrmMethodsTree" name="methods_tree" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frameAbstract">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <property name="spacing">
             <number>4</number>
            </property>
            <item>
             <spacer name="verticalSpacer_4">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QToolButton" name="tbLinkMethod">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Добавить метод</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="zrm_data.qrc">
                <normaloff>:/data/icons/left.png</normaloff>:/data/icons/left.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="tbUnlinkMethod">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Убрать метод</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="zrm_data.qrc">
                <normaloff>:/data/icons/delete.png</normaloff>:/data/icons/delete.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_5">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="ZrmMethodsTree" name="methods_abstract" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <action name="actApply">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/apply.png</normaloff>:/data/icons/apply.png</iconset>
   </property>
   <property name="text">
    <string>Применить</string>
   </property>
   <property name="toolTip">
    <string>Применить (Ctrl+Return)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Return</string>
   </property>
  </action>
  <action name="actUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/undo.png</normaloff>:/data/icons/undo.png</iconset>
   </property>
   <property name="text">
    <string>Отмена</string>
   </property>
   <property name="toolTip">
    <string>Отмена изменений (Ctrl+Backspace)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Backspace</string>
   </property>
  </action>
  <action name="actDelete">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/delete.png</normaloff>:/data/icons/delete.png</iconset>
   </property>
   <property name="text">
    <string>Удалить</string>
   </property>
   <property name="toolTip">
    <string>Удалить (Ctrl+Del)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Del</string>
   </property>
  </action>
  <action name="actLink">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/link.png</normaloff>:/data/icons/link.png</iconset>
   </property>
   <property name="text">
    <string>Связать</string>
   </property>
   <property name="toolTip">
    <string>Связать (Ctrl+L)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actNew">
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/new_group.png</normaloff>:/data/icons/new_group.png</iconset>
   </property>
   <property name="text">
    <string>Новая категория</string>
   </property>
  </action>
  <action name="actNewChild">
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/new.png</normaloff>:/data/icons/new.png</iconset>
   </property>
   <property name="text">
    <string>Создать </string>
   </property>
   <property name="toolTip">
    <string>Создать(Ctrl+N)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actMethodEdit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/edit.png</normaloff>:/data/icons/edit.png</iconset>
   </property>
   <property name="text">
    <string>Редактировать метод</string>
   </property>
   <property name="toolTip">
    <string>Редактировать метод (Ctrl+E)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actAllMethods">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/all_methods.png</normaloff>:/data/icons/all_methods.png</iconset>
   </property>
   <property name="text">
    <string>Все методы</string>
   </property>
   <property name="toolTip">
    <string>Все методы(Ctrl+A)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actCopyModel">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/new_akb.png</normaloff>:/data/icons/new_akb.png</iconset>
   </property>
   <property name="text">
    <string>Создать копию</string>
   </property>
   <property name="toolTip">
    <string>Создать копию (Ctrl+C)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actUnload">
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/unload.png</normaloff>:/data/icons/unload.png</iconset>
   </property>
   <property name="text">
    <string>Выгрузить методы</string>
   </property>
   <property name="toolTip">
    <string>Выгрузить файл БД с методами</string>
   </property>
  </action>
  <action name="actLoad">
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/load.png</normaloff>:/data/icons/load.png</iconset>
   </property>
   <property name="text">
    <string>Загрузить методы</string>
   </property>
   <property name="toolTip">
    <string>Загрузить новый файл БД с методами</string>
   </property>
  </action>
  <action name="actCopyMethod">
   <property name="icon">
    <iconset resource="zrm_data.qrc">
     <normaloff>:/data/icons/copy_method.png</normaloff>:/data/icons/copy_method.png</iconset>
   </property>
   <property name="text">
    <string>Создать копию метода</string>
   </property>
   <property name="toolTip">
    <string>Создать копию метода</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ZrmMethodsTree</class>
   <extends>QWidget</extends>
   <header>zrmmethodstree.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ZrmStagesEditor</class>
   <extends>QWidget</extends>
   <header>zrmstageseditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tbLinkMethod</tabstop>
  <tabstop>tbUnlinkMethod</tabstop>
 </tabstops>
 <resources>
  <include location="zrm_data.qrc"/>
 </resources>
 <connections/>
</ui>
